### ISSUE TITLE
`GCRYPT_1.2' not found on Debian unstable

### ISSUE THREAD

**Issue by kwbr** [leamas/spotify-make#40]
With the current Debian unstable the same error as on Fedora (see #33) appears:

```
$ spotify 
/home/kai/.local/lib/spotify-client/spotify: /home/kai/.local/lib/spotify-client/libgcrypt.so.11: version `GCRYPT_1.2' not found (required by /home/kai/.local/lib/spotify-client/libcef.so)
```

**Comment by leamas** [comment]
Hm... In order to do this I need toi create a virtual suse host running zypper. As of now, I'm not motivated enough to do this.

What's needed is

  - The 64-bit library as of above.
  - The corresponding 32-bit library.
  -  The license file from any the 64- or 32-bit package (they should be the same).
  - The output from ```lsb-release``` on a suse host.

If someone could provide these one way or another it should be able to update spotify-make to cope with the situation.

**Comment by SecUpwN** [comment]
@leamas, first of all: Happy New Year! Here we rock:

* The [64-bit library](http://download.opensuse.org/repositories/home:/fstrba/openSUSE_13.2/x86_64/libgcrypt11-1.5.4-1.1.x86_64.rpm) as of above.
* The corresponding [32-bit library](http://download.opensuse.org/repositories/home:/fstrba/openSUSE_13.2/i586/libgcrypt11-1.5.4-1.1.i586.rpm).
* The [license file](https://www.gnu.org/software/libgcrypt/) from any the 64- or 32-bit package (they should be the same).
* The output from `lsb-release -i` on a suse host. -> `Distributor ID: openSUSE project`

**Comment by leamas** [comment]
Looks good besides  the license file. That file is not the license but rather the README. It doesn't even include a license statement.

Since this is a GNU  package, there should be a COPYING file somewhere. Can you fix this?

**Comment by SecUpwN** [comment]
>Since this is a GNU package, there should be a COPYING file somewhere. Can you fix this?

I think so. The file in question is located in `/usr/share/doc/packages/libgcrypt/COPYING` and is just a copy of [GPL v2](http://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgcrypt.git;a=blob;f=COPYING;h=d60c31a97a544b53039088d14fe9114583c0efc3;hb=1cfac9fd5f628c2d505b6c01d2cc0657840299ed) should be the solution to your question..

**Comment by leamas** [comment]
No, it's not a copy - there is some boilerplate text filled in. I need the real /usr/share/doc/packages/libgcrypt/COPYING

**Comment by SecUpwN** [comment]
>I need the real `/usr/share/doc/packages/libgcrypt/COPYING

Ok, fair. There are actually two different ones: `libgcrypt11` has `GNU GENERAL PUBLIC LICENSE` and `libgcrypt20` uses `GNU LESSER GENERAL PUBLIC LICENSE`. So which one do you need?

**Comment by leamas** [comment]
The file related to libgrypt11

**Comment by SecUpwN** [comment]
>The file related to libgrypt11

[There](http://wikisend.com/download/860946/COPYING) ya go, the download link will be valid for 7 days.

**Comment by leamas** [comment]
OK, I pushed a new feature branch named 'issue-40'. If anyone could test it on SUSE it would be really nice...

**Comment by SecUpwN** [comment]
Thanks! Calling @rriemann to do this, My internet connection currently does not allow to do so.

**Comment by kwbr** [comment]
Does not work for me on Debian unstable. It seems libgcrypt.so.20 is linked as libgcrypt.so.11. But this library does not provide  GCRYPT_1.2

```
$ ls -l /home/kai/.local/lib/spotify-client/libgcrypt.so.11
lrwxrwxrwx 1 kai kai 37 Feb 10 16:40 /home/kai/.local/lib/spotify-client/libgcrypt.so.11 -> /lib/x86_64-linux-gnu/libgcrypt.so.20
$ spotify 
/home/kai/.local/lib/spotify-client/spotify: /home/kai/.local/lib/spotify-client/libgcrypt.so.11: version `GCRYPT_1.2' not found (required by /home/kai/.local/lib/spotify-client/libcef.so)
```

**Comment by leamas** [comment]
That link looks bogus to me, for the reason you describe.

This could be patched easily in the same way as for many other distros if you could provide a libgrypt.so.1.1 built for unstable; there is probably some compatibility  package out there (?)

**Comment by SecUpwN** [comment]
Gosh, how I hate that people do not follow up on their own Issues! :(
[Here](http://ftp.de.debian.org/debian/pool/main/libg/libgcrypt20/libgcrypt11-dev_1.5.4-3+really1.6.2-4_all.deb) ya go, @leamas. Hope that helps and you'll merge this now.

**Comment by Morganlej** [comment]
Thanks for your contribution :)
People have lives and other stuff too, you know.

**Comment by SecUpwN** [comment]
@leamas, would you please merge this now? I have the same error and think this Issue may fix it.

```
/home/SecUpwN/.local/lib/spotify-client/spotify: /lib64/libcrypto.so.1.0.0: no version information available (required by /home/SecUpwN/.local/lib/spotify-client/spotify)
/home/SecUpwN/.local/lib/spotify-client/spotify: /lib64/libssl.so.1.0.0: no version information available (required by /home/SecUpwN/.local/lib/spotify-client/spotify)
/home/SecUpwN/.local/lib/spotify-client/spotify: /home/SecUpwN/.local/lib/spotify-client/libgcrypt.so.11: version `GCRYPT_1.2' not found (required by /home/SecUpwN/.local/lib/spotify-client/libcef.so)
```

**Comment by jkudish** [comment]
the issue-40 branch breaks the installer in Fedora 22 for me, fwiw.

**Comment by SecUpwN** [comment]
> the issue-40 branch breaks the installer in Fedora 22 for me, fwiw.

@jkudish, please submit the files listed in https://github.com/leamas/spotify-make/issues/40#issuecomment-68492903 so that @leamas can have a look.

**Comment by SecUpwN** [comment]
@aspiers, I am now using your wonderful [openSUSE Spotify Installer](https://github.com/aspiers/opensuse-spotify-installer/) in conjunction with my own project that I'd like you to join (or at least give a star): The [Spotify-AdKiller](https://github.com/SecUpwN/Spotify-AdKiller/)! Are you able to help @leamas to solve this Issue? It is also preventing successful launchg of `my-spotify` on openSUSE (currently using `Tumbleweed`). Thanks for developing your installer for my beloved openSUSE system!

**Comment by SecUpwN** [comment]
Listen up folks, please test if https://github.com/aspiers/opensuse-spotify-installer/issues/42#issuecomment-211305594 is the solution to this!
